app/src/main/java/com/khmelenko/lab/travisclient/widget/BuildView.java

```diff
@@ -136,10 +108,9 @@ private void setBuildDuration(long durationInMillis) {
             String duration = TimeConverter.durationToString(durationInMillis);
             duration = getContext().getString(R.string.build_duration, duration);
             mDuration.setText(duration);
+            mDuration.setVisibility(VISIBLE);
         } else {
-            String stateInProgress = getContext().getString(R.string.build_build_in_progress);
-            String duration = getContext().getString(R.string.build_duration, stateInProgress);
-            mDuration.setText(duration);
+            mDuration.setVisibility(GONE);
```

Comment:
The information that the build duration is "in progress" and to show "finished at" when the build has not yet finished feels wrong to me.
Additionally the build status already indicated that the build is in progress, which in turn means displaying the same information three times in one view.